File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int len = id_list.length;
        int[] answer = new int[len];
        int[] rep_cnt = new int[len];
        boolean[][] send = new boolean[len][len];
        Map<String, Integer> user = new HashMap<>();
        
        
        Arrays.fill(answer,0);
        Arrays.fill(rep_cnt, 0);
        for(int i=0; i<len; i++){
            Arrays.fill(send[i], false);
        }
        for(int i=0; i<len; i++){
            user.put(id_list[i], i);
        }
        
        
        for(int i=0; i<report.length; i++){
            String[] rep = report[i].split(" ");
            if(!send[user.get(rep[1])][user.get(rep[0])])  rep_cnt[user.get(rep[1])]++;
            send[user.get(rep[1])][user.get(rep[0])] = true;
        }
        
        
        for(int i=0; i<len; i++){
            if(rep_cnt[i]>=k){
                for(int j=0; j<len; j++){
                    if(send[i][j]){
                        answer[j]++;
                    }
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    var answer = new Array(id_list.length).fill(0);
    var idIndex = id_list.reduce((acc, cur, idx) => {
        acc[cur] = idx;
        return acc
    }, {});

    var reportId = {}
    report.forEach(str => {
        var [from, to] = str.split(" ");
        reportId[to] ? reportId[to].add(from) : reportId[to] = new Set().add(from)
    })
    // console.log(fromId);
    for(id in reportId) {
        if(reportId[id].size >= k) {
            // console.log(id)
            for(fromId of reportId[id]){
                answer[idIndex[fromId]]++;
            }
        }
    }
    return answer;
}